webxr-api/input.rs

```diff
@@ -0,0 +1,39 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+use crate::Input;
+use crate::Native;
+
+use euclid::TypedRigidTransform3D;
+
+#[derive(Copy, Clone, Debug)]
+#[cfg_attr(feature = "ipc", derive(serde::Serialize, serde::Deserialize))]
+pub enum Handedness {
+    None,
+    Left,
+    Right,
+}
+
+#[derive(Copy, Clone, Debug)]
+#[cfg_attr(feature = "ipc", derive(serde::Serialize, serde::Deserialize))]
+pub enum TargetRayMode {
+    Gaze,
+    TrackedPointer,
+    Screen,
+}
+
+#[derive(Copy, Clone, Debug)]
+#[cfg_attr(feature = "ipc", derive(serde::Serialize, serde::Deserialize))]
+pub struct InputInfo {
+    pub handedness: Handedness,
+    pub target_ray_mode: TargetRayMode,
+    pub id: u32,
```

Comment:
Oh joy, ids. Can we at least declare a type synonym `InputId`?

Reply:
The spec actually exposes these :smile: . It's part of gamepad but also used in input.

Reply:
Oh yuk, not even with a type synonym?

Reply:
I added one.